src/storage-adapters/storage-adapter.ts

```diff
@@ -0,0 +1,16 @@
+export interface FileRef {
+	filename: string;
+	href: string;
+	storage: string;
+}
+
+export interface StorageAdapter {
+	name: string;
+	saveFile(filename: string, data: Buffer): Promise<FileRef>;
+	getFileData(fileRef: FileRef): Promise<Buffer>;
+	deleteFile(fileRef: FileRef): Promise<void>;
```

Comment:
We may need to know if the deleteFile has worked, right? Should the interface model this with a return value other than void? When we have a data model delete and we may not be able to delete the storageADapter resource, we may not delete the parant object. Otherwise, we will loose the entry to the storage to retry the delete?

Reply:
About the method: if should throw an Error if not successful. I can add this to the interface.

But you raise an important point, because this is a missing feature, meaning that I haven't hooked up yet the delete behavior. In fact I can't find a place in sbvr-types to handle the delete. I only have two functions, `validate` where we create the S3 object, and `fetchProcessing` which just returns the value.

But there's no `delete` function that I can hook on on this repo. 

Do you know where that function could be?

Reply:
@Page- What is the expected behavior of sbvr-types when the delete of a webresource entry fails? The entry in the DB shouldn't be deleted to not loose the minio/s3 resource URL, but the sbvr-types don't have a procedure for checking on delete.

Reply:
This will be addressed from pinejs; deleting a resource will remove the object from S3